src/dolphin/timeseries.py

```diff
@@ -312,21 +313,77 @@ def write_callback(fut: Future):
 def create_velocity(
     unw_file_list: Sequence[PathOrStr],
     output_file: PathOrStr,
+    reference: ReferencePoint,
     date_list: Sequence[DateOrDatetime] | None = None,
     cor_file_list: Sequence[PathOrStr] | None = None,
     cor_threshold: float = 0.2,
     block_shape: tuple[int, int] = (512, 512),
     num_threads: int = 5,
+    add_overviews: bool = True,
 ) -> None:
-    """Perform pixel-wise (weighted) linear regression to estimate velocity."""
+    """Perform pixel-wise (weighted) linear regression to estimate velocity.
+
+    Parameters
+    ----------
+    unw_file_list : Sequence[PathOrStr]
+        List of unwrapped phase files.
+    output_file : PathOrStr
+        The output file to save the velocity to.
+    reference : ReferencePoint
+        The (row, col) to use as reference before fitting the velocity.
+        This point will be subtracted from all other points before solving.
+    date_list : Sequence[DateOrDatetime], optional
+        List of dates corresponding to the unwrapped phase files.
+        If not provided, will be parsed from filenames in `unw_file_list`.
+    cor_file_list : Sequence[PathOrStr], optional
+        List of correlation files to use for weighting the velocity estimation.
+        If not provided, all weights are set to 1.
+    cor_threshold : float, optional
+        The correlation threshold to use for weighting the velocity estimation.
+        Default is 0.2.
+    block_shape : tuple[int, int], optional
+        The shape of the blocks to process in parallel.
+        Default is (512, 512)
+    num_threads : int, optional
+        The parallel blocks to process at once.
+        Default is 5.
+    add_overviews : bool, optional
+        If True, creates overviews of the new velocity raster.
+        Default is True.
+
+    """
+    if Path(output_file).exists():
+        logger.info(f"Output file {output_file} already exists, skipping velocity")
+        return
+
     if date_list is None:
         date_list = [get_dates(f)[1] for f in unw_file_list]
-        # TODO: need the relative one?
     x_arr = datetime_to_float(date_list)
 
+    # Set up the input data readers
+    out_dir = Path(output_file).parent
+    unw_reader = io.VRTStack(
+        file_list=unw_file_list,
+        outfile=out_dir / "velocity_inputs.vrt",
+        skip_size_check=True,
+    )
+    if cor_file_list is not None and len(cor_file_list) == len(unw_file_list):
+        cor_reader = io.VRTStack(
+            file_list=cor_file_list,
+            outfile=out_dir / "cor_inputs.vrt",
+            skip_size_check=True,
```

Comment:
So if the user wants to weight the estimation with correlation but they have a bug in the input, I think it is better to throw an exception instead of silently ignoring their request. What do you think?

Reply:
that's a good question... the `skip_size_check` only checks that all files in `file_list` have the same x/y size: https://github.com/isce-framework/dolphin/blob/39ea9944034344195239bbffbc58f6aaf07beb74/src/dolphin/io/_readers.py#L738-L739
I'm not sure what the most likely user error with the input is here, but if you think it's not worth a time saving I'll turn on this check.

The initial reason I had the skip was that it saves a decent amount of time (a couple minutes) if the stack size is large and all the files are on S3, since the check goes and opens each file with GDAL.
For local files, it's only a second or two for this assertion.

Reply:
Oh sure, yes I like that idea of raising an error if, e.g., they pass 20 unw files and 17 cor files

Reply:
I changed it, but then realized why I had it skipping- after an inversion, we can't pass the correlation files to weight the velocity fit since there's more correlation and (inverted) phase rasters.
I still think you're right about it erroring in this step, so I changed the displacement part to pass `None` to `cor_file_list` if we did a network inversion